sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,365 @@
+
+    """Collects database information and schemas and submits to the agent's queue as dictionaries
+    schema dict
+    key/value:
+        "name": str
+        "id": str
+        "owner_name": str
+        "tables" : list of tables dicts
+            table
+            key/value:
+                "id" : str
+                "name" : str
+                columns: list of columns dicts
+                    columns
+                    key/value:
+                        "name": str
+                        "data_type": str
+                        "default": str
+                        "nullable": bool
+            indexes : list of index dicts
+                index
+                key/value:
+                    "name": str
+                    "type": str
+                    "is_unique": bool
+                    "is_primary_key": bool
+                    "is_unique_constraint": bool
+                    "is_disabled": bool,
+                    "column_names": str
+            foreign_keys : list of foreign key dicts
+                foreign_key
+                key/value:
+                    "foreign_key_name": str
+                    "referencing_table": str
+                    "referencing_column": str
+                    "referenced_table": str
+                    "referenced_column": str
+            partitions: partition dict
+                partition
+                key/value:
+                    "partition_count": int
+    """
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def collect_schemas_data(self):
+        self._dataSubmitter.reset()
+        self._dataSubmitter.set_base_event_data(
+            self._check.resolved_hostname,
+            self._check.non_internal_tags,
+            self._check._config.cloud_metadata,
+            "{},{}".format(
+                self._check.static_info_cache.get(STATIC_INFO_VERSION, ""),
+                self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, ""),
+            ),
+        )
+
+        # returns if to stop, True means stop iterating.
+        def fetch_schema_data(cursor, db_name):
+            db_info = self._query_db_information(db_name, cursor)
+            schemas = self._query_schema_information(cursor)
+            self._dataSubmitter.store_db_info(db_name, db_info)
+            for schema in schemas:
+                tables = self._get_tables(schema, cursor)
+                tables_chunks = list(get_list_chunks(tables, self.TABLES_CHUNK_SIZE))
+                for tables_chunk in tables_chunks:
+                    if self._dataSubmitter.exceeded_total_columns_number():
+                        # TODO Report truncation to the backend
+                        self._log.warning(
+                            "Truncated data due to the max limit, stopped on db - {} on schema {}".format(
+                                db_name, schema["name"]
+                            )
+                        )
+                        return True
```

Comment:
Use True/False boolean to indicate stop iteration could be confusing, raise a `StopIteration` exception or a custom `MaxColumnCountReached` exception might be easier to understand